### PR TITLE
Allow <> through service extension property validation.

### LINKED ISSUE #214: Input character validation to allow "<>" in endpoint extension properties
Allow "<>" through input validation of extension properties to enable the encoding of multiple host DNs - to be used by a single service offered on multiple hosts (each with unique Subject DN and shared service SubjectAltName) behind load-balanced URL. e.g. -
image

Ref: RFC1779

### PR DESCRIPTION
**Pull request by ineilson**
Including basic ServiceService extension property input validation tests. Fixes #214

### CODE REVIEW

**Comment by gregcorbett** [comment]
Have the argus folk confirmed they can parse the escaped "less than" and "greater than" symbols?

**Comment by ineilson** [comment]
Well, not exactly, claim: "could deal with any format" -

> I haven't had time to try to really think about the use of &lt; and &gt; to separate the values.
> The parser I have could indeed deal with any format, but I am not too sure:
> - If it would impact other consumers of gocdb data (I don't know if anyone else reads this)
> - If there are no "standard" for such thing
> The good thing is that I would not expect '&gt;' to be part of a valid DN, so we wouldn't have to deal with escaping the DN.

For the above bullet points we have already discussed and decided to go ahead. For further testing I thought we were going to put this on a/the test server and advertise the change for people to know and check. There is also a plan to raise this as part of the next security team meeting.

File: tests/unit/lib/Gocdb_Services/ServiceServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  /**
+   * Check the basics -
+   * Duplicates some simple testing from ExtensionsTest
+   */
+  public function testValidateProperty(){
+    print __METHOD__ . "\n";
+
+    list ($service, $user, $serviceService) = $this->createValidationEntities();
+   
+    $values[0] = array("ValidName1","ValidValue");
```
Comment (gregcorbett) [inline tests/unit/lib/Gocdb_Services/ServiceServiceTest.php:149]: are these arrays of the form ("Property Name", Property Value")?

File: tests/unit/lib/Gocdb_Services/ServiceServiceTest.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+   * Check that validation of property name is operating as expected
+   * Added to test code changes to pass through "<>" chars
+   * @depends testValidateProperty
+   */
+  public function testValidatePropertyNameFails(){
+    print __METHOD__ . "\n";
+
+    list ($service, $user, $serviceService) = $this->createValidationEntities();
+   
+    $values[0] = array("<Invalid>","<valid>");
```
Comment (gregcorbett) [inline tests/unit/lib/Gocdb_Services/ServiceServiceTest.php:172]: Is this array of the form ("Property Name", Property Value")? Its worth a comment as to why this case is invalid.

File: tests/unit/lib/Gocdb_Services/ServiceServiceTest.php [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  /**
+   * Check that validation of property value is operating as expected
+   * @depends testValidateProperty
+   */
+  public function testValidatePropertyValueFails(){
+    print __METHOD__ . "\n";
+
+    list ($service, $user, $serviceService) = $this->createValidationEntities();
+   
+    $values[0] = array("Valid","'Not Valid'");
```
Comment (gregcorbett) [inline tests/unit/lib/Gocdb_Services/ServiceServiceTest.php:188]: Is this array of the form ("Property Name", Property Value")? Its worth a comment as to why this case is invalid.

File: tests/unit/lib/Gocdb_Services/ServiceServiceTest.php (hunk 1 above)
Comment (ineilson) [reply tests/unit/lib/Gocdb_Services/ServiceServiceTest.php:149]: Yes.

File: tests/unit/lib/Gocdb_Services/ServiceServiceTest.php (hunk 2 above)
Comment (ineilson) [reply tests/unit/lib/Gocdb_Services/ServiceServiceTest.php:172]: Added comment

File: tests/unit/lib/Gocdb_Services/ServiceServiceTest.php (hunk 3 above)
Comment (ineilson) [reply tests/unit/lib/Gocdb_Services/ServiceServiceTest.php:188]: Added comment

**Review by gregcorbett — DISMISSED** [review]

**Comment by gregcorbett** [comment]
@ineilson, could this be rebased and the conflicts fixed?

**Review by gregcorbett — APPROVED** [review]

**Pull request merged**